src/layout/mod.rs

```diff
@@ -1335,6 +1335,39 @@ impl<W: LayoutElement> Layout<W> {
         }
     }
 
+    pub fn move_workspace_to_output(&mut self, output: &Output) {
+        let MonitorSet::Normal {
+            monitors,
+            active_monitor_idx,
+            ..
+        } = &mut self.monitor_set
+        else {
+            return;
+        };
+
+        let current = &mut monitors[*active_monitor_idx];
+        if current.active_workspace_idx == current.workspaces.len() - 1 {
+            // Insert a new empty workspace.
+            let ws = Workspace::new(current.output.clone(), current.options.clone());
+            current.workspaces.push(ws);
+        }
+        let mut ws = current.workspaces.remove(current.active_workspace_idx);
+        current.active_workspace_idx = current.active_workspace_idx.saturating_sub(1);
+
+        let view_size = output_size(output);
+        let working_area = compute_working_area(output, self.options.struts);
+        ws.set_view_size(view_size, working_area);
```

Comment:
Oh yeah and you need to update `ws.output` and `ws.original_output` to `output`. That should fix the failing tests.

Reply:
Would `ws.set_output(Some(output.clone())` work? It would take care of computing the working area and also sends output leave and enter events.

Reply:
Yes, that's even better. I guess you could check the code that adds or removes monitors because that's the other place that moves workspaces between monitors.

You will still need to update `ws.original_output` though, since that code doesn't (and shouldn't), but this code should, since the user explicitly asked the workspace to move to a different output.

Reply:
Changed and several test runs with `RUN_SLOW_TESTS=1` are happy locally